client/src/main/java/io/micrometer/prometheus/rsocket/PrometheusRSocketClient.java

```diff
@@ -150,15 +156,57 @@ public void close() {
     }
   }
 
+  /**
+   * Pushes the data in a blocking way and closes the connection.
+   */
+  public void pushAndCloseBlockingly() {
+    pushAndCloseBlockingly(Duration.ofSeconds(5));
+  }
+
+  /**
+   * Pushes the data in a blocking way and closes the connection.
+   * @param timeout the amount of time to wait for the data to be sent
+   */
+  public void pushAndCloseBlockingly(Duration timeout) {
+    CountDownLatch latch = new CountDownLatch(1);
+    PublicKey key = latestKey.get();
+    if (key != null) {
+      try {
+        sendingSocket
+            .requestResponse(scrapePayload(key))
+            .doOnEach(signal -> latch.countDown())
```

Comment:
Nit pick: Since it is a mono, it's almost the same thing, but wouldn't `doFinally` be more semantically what we want? I think that would cover cancellation also. If we want to do something other than count down the latch on cancellation, we should probably add logic for that.